db_stress_tool/db_stress_tool.cc

```diff
@@ -362,11 +362,11 @@ int db_stress_tool(int argc, char** argv) {
   // Initialize the Zipfian pre-calculated array
   InitializeHotKeyGenerator(FLAGS_hot_key_alpha);
   shared.reset(new SharedState(db_stress_env, stress.get()));
-  if (RunStressTest(shared.get())) {
-    return 0;
-  } else {
-    return 1;
-  }
+  bool run_stress_test = RunStressTest(shared.get());
+  // Close DB in CleanUp() before destructor to prevent race between destructor
+  // and operations in listener callbacks (e.g. MultiOpsTxnsStressListener).
+  stress->CleanUp();
```

Comment:
Does that mean DB's Close() called in DB's destructor will still race with access of db_ pointer even though the destructor hasn't existed and db_pointer is not nullptr yet at that point?

Reply:
If we call Close() explicitly, DB destructor should not call DB Close() again.